src/fairseq2/recipes/lm/_online_finetune/_common.py

```diff
@@ -302,23 +301,33 @@ def find_first_value(lst, value):
 
 
 def generate_rollouts(
-    prompts: List[List[int]], dp_gang: Gang, vllm_model, sampling_params=None
+    prompts: List[List[int]],
+    dp_gang,
+    vllm_model,
+    operation: str = "rollout",
+    sampling_params=None,
 ):
+    if operation not in ["rollout", "reward"]:
+        raise ValueError("Operation must be either 'rollout' or 'reward'.")
+
     prompts_to_generate = [None] * dp_gang.size
     if dp_gang.rank == 0:
         dp_gang.gather_object(prompts, prompts_to_generate, 0)
     else:
         dp_gang.gather_object(prompts, None, 0)
-
     if dp_gang.rank == 0:
         rank_batch_sizes = [len(l) for l in prompts_to_generate]
         flat_request_list = []
         for rank_prompts in prompts_to_generate:
             flat_request_list.extend(rank_prompts)
 
-        rollouts = vllm_model.rollout_from_model(
-            flat_request_list, sampling_params=sampling_params
-        )
+        if operation == "rollout":
+            rollouts = vllm_model.rollout_from_model(
+                flat_request_list, sampling_params=sampling_params
+            )
+
+        elif operation == "reward":
+            rollouts = vllm_model.reward_from_model(flat_request_list)
```

Comment:
I wanted to keep both of these calls in the same function generate_rollouts() function, but not sure if this is the best way to do it.